src/rgw/rgw_dmclock_async_scheduler.cc

```diff
@@ -41,20 +42,21 @@ int AsyncScheduler::schedule_request_impl(const client_id& client,
                                           const Time& time, const Cost& cost,
                                           optional_yield yield_ctx)
 {
-    ceph_assert(yield_ctx);
-
+#ifdef HAVE_BOOST_CONTEXT
+  if (yield_ctx) {
     auto &yield = yield_ctx.get_yield_context();
     boost::system::error_code ec;
     async_request(client, params, time, cost, yield[ec]);
-
     if (ec){
       if (ec == boost::system::errc::resource_unavailable_try_again)
         return -EAGAIN;
       else
         return -ec.value();
     }
 
-    return 0;
+  }
+#endif
+  return 0;
```

Comment:
i'm not sure that returning success is the right thing to do here.. if the only dependency on AsyncScheduler is coming from the beast frontend, maybe we should move rgw_dmclock_async_scheduler.cc into this block with the frontend code:
```cmake
if(WITH_RADOSGW_BEAST_FRONTEND)
  list(APPEND radosgw_srcs
    rgw_asio_client.cc
    rgw_asio_frontend.cc)
endif()
```
WITH_RADOSGW_BEAST_FRONTEND is conditional on WITH_BOOST_CONTEXT so that should do the right thing?